source/grammar/qasm3.g4

```diff
@@ -1,610 +0,0 @@
-/***** ANTLRv4  grammar for OpenQASM3.0. *****/
-
-grammar qasm3;
-
-/**** Parser grammar ****/
-
-program
-    : header (globalStatement | statement)*
-    ;
-
-header
-    : version? include* io*
-    ;
-
-version
-    : 'OPENQASM' ( Integer | RealNumber ) SEMICOLON
-    ;
-
-include
-    : 'include' StringLiteral SEMICOLON
-    ;
-
-ioIdentifier
-    : 'input'
-    | 'output'
-    ;
-io
-    : ioIdentifier classicalType Identifier SEMICOLON
-    ;
-
-globalStatement
-    : subroutineDefinition
-    | externDeclaration
-    | quantumGateDefinition
-    | calibration
-    | quantumDeclarationStatement  // qubits are declared globally
-    | pragma
-    ;
-
-statement
-    : expressionStatement
-    | assignmentStatement
-    | classicalDeclarationStatement
-    | branchingStatement
-    | loopStatement
-    | endStatement
```

Comment:
Just want to have a clarification, what is the purpose of endStatement in the grammar? Is it similar to Fortran's `end`?

Reply:
Yeah, it does the same thing: ends the programme.  It's buried in a slightly arcane position in the spec, just before the last example of the looping and branching section: https://qiskit.github.io/openqasm/language/classical.html#looping-and-branching.